api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -105,6 +105,16 @@ const (
 	NetworkPluginModeOverlay NetworkPluginMode = "overlay"
 )
 
+// NetworkDataplaneType is the type of network dataplane to use.
+type NetworkDataplaneType string
+
+const (
+	// NetworkDataplaneTypeAzure ...
```

Comment:
Could you flesh out these comments? I know the variable name itself is self-descriptive, but I prefer boring, prosaic comments to an ellipsis. (I know we do this elsewhere, but IMHO it's like fooling the doc linter.)